### PR TITLE
feat(cli) - static request & response auto-writing

### PR DESCRIPTION
**Pull request by ttodua**
to test out, run:
`npm run cli.ts gate fetchTrades BTC/USDT undefined 5 -- -- --response --name "My description"`

and then see the diff of static.json changes

### CODE REVIEW

**Comment by carlosmiei** [comment]
@ttodua can you update the `cli.js` as well?

File: examples/js/cli.js [hunk 1]
```diff
@@ -237,7 +256,8 @@ const printHumanReadable = (exchange, result) => {
                     return String (x)
                 }
             })
-            log (result.length > 0 ? configuredAsTable (result.map (element => {
+            log (result.length > 0 ? configuredAsTable (result.map (rawElement => {
+                const element = Object.assign ({}, rawElement)
```
Comment (ttodua) [inline examples/js/cli.js:260]: this edit was missing from previous PR merger

**Comment by ttodua** [comment]
> @ttodua can you update the `cli.js` as well?

for some reason I couldnt make that import inside .js, if you can try on your end, something weird on my side.

**Comment by carlosmiei** [comment]
@ttodua Sorry, which import

**Comment by ttodua** [comment]
> @ttodua Sorry, which import

 line 8th in cli.ts, but can't make it work in cli.js. if you have a minute to test it out if you know why that happens

**Comment by carlosmiei** [comment]
@ttodua will try to test it

**Comment by carlosmiei** [comment]
found a small issue when trying to provide for instance `--no-keys` breaks the parsing of the args

```
 ✗ npm run cli.ts binance fetchTrades BTC/USDT undefined 5 -- -- --no-keys --response --name "My description"

binance.fetchTrades (BTC/USDT, , 5, My description)
BadRequest binance {"code":-1101,"msg":"Too many parameters; expected '5' and received '16'."}
---------------------------------------------------
[BadRequest] binance {"code":-1101,"msg":"Too many parameters; expected '5' and received '16'."}

    at throwExactlyMatchedException  ts/src/base/Exchange.ts:4889          throw new exact[string] (message);                                              
    at handleErrors                  ts/src/binance.ts:11650               this.throwExactlyMatchedException (this.exceptions['exact'], error, feedback);  
    at <anonymous>                   ts/src/base/Exchange.ts:1002          const skipFurtherErrorHandling = this.handleErrors (response.status, response.s…
    at processTicksAndRejections     node:internal/process/task_queues:95                                                                                  
    at fetch2                        ts/src/base/Exchange.ts:4391          return await this.fetch (request['url'], request['method'], request['headers'],…
    at request                       ts/src/binance.ts:11692               const response = await this.fetch2 (path, api, method, params, headers, body, c…
    at fetchTrades                   ts/src/binance.ts:4848                response = await this.publicGetAggTrades (this.extend (request, params));       
    at run                           examples/ts/cli.ts:370                const result = await exchange[methodName] (... args)                            

binance {"code":-1101,"msg":"Too many parameters; expected '5' and received '16'."}
```

**Comment by carlosmiei** [comment]
@ttodua I'm still not sure which import is not working in the cli.js

**Comment by ttodua** [comment]
> @ttodua I'm still not sure which import is not working in the cli.js

so you have been able to run cli.js with this commit? https://github.com/ccxt/ccxt/blob/44118ca1005e4957338487e048d3607c0bf9dd7f/examples/js/cli.js

**Comment by ttodua** [comment]
ah, sorry, i had reverted that cli.js update back, and i was still referring to that cli.js which was no longer present (because reverted) in PR current files. so i've linked the commit which was counterpart update of .ts , but that .js didnt work. please lmk if you able to run that cli.js

**Comment by carlosmiei** [comment]
@ttodua I think so, cli.js seems to be working normally on my end with this PR

**Comment by ttodua** [comment]
> > @ttodua I'm still not sure which import is not working in the cli.js
> 
> so you have been able to run cli.js with this commit? [`44118ca`/examples/js/cli.js](https://github.com/ccxt/ccxt/blob/44118ca1005e4957338487e048d3607c0bf9dd7f/examples/js/cli.js)

so, i will return back this commit if that works on your side (though it didnt on my side).

**Comment by ttodua** [comment]
@carlosmiei pushed an update , also fixed response<>request overwrite glitch

**Comment by ttodua** [comment]
travis passes

**Comment by carlosmiei** [comment]
@ttodua yeah seems to be working fine now

```
npm run cli.js gate fetchTrades BTC/USDT undefined 5 -- -- --no-keys --request --name "My description"



> ccxt@4.4.25 cli.js
> node ./examples/js/cli.js gate fetchTrades BTC/USDT undefined 5 -- --no-keys --request --name My description

2024-11-06T10:23:31.863Z
Node.js: v18.18.0
CCXT v4.4.25
gate.fetchTrades (BTC/USDT, , 5, My description)
2024-11-06T10:23:37.344Z iteration 0 passed in 347 ms

```
image

**Pull request merged**